lua/niknaks/modules/sh_bitbuffer.lua

```diff
@@ -242,132 +261,157 @@ do
 
 	-- B |00000000|00000000|00000000|00000000|
 	local b_mask = 0xFFFFFFFF
-	local bswap, rshift, band, brshift, source = bswap, rshift, band, brshift, source
+
+	--- @param int number
+	--- @param bits number
+	--- @return number
 	local function swap( int, bits )
 		return brshift( bswap( int ), 32 - bits )
 	end
+
+	--- @param self BitBuffer
+	--- @param int number
+	--- @param bits number
+	--- @return self BitBuffer The BitBuffer that was modified
 	function writeraw( self, int, bits )
 		if self._little_endian and bits % 8 == 0 then
 			int = swap( int, bits )
 		end
-		local rshift, bor, band, max, lshift = rshift, bor, band, max, lshift
+
 		local tell = self._tell
 		self._tell = tell + bits
 		self._len = max( self._len, self._tell )
+
 		-- Retrive data pos
 		local i_word = rshift( tell, 5 ) + 1  -- [ 1 - length ]
 		local bitPos = tell % 32 -- [[ 0 - 31 ]]
 		local ebitPos = bitPos + bits -- The end bit pos
+
 		-- DataMask & Data
 		local mask = bor( lshift( b_mask, 32 - bitPos ), rshift( b_mask, ebitPos ) )
 		local data = band( self._data[ i_word ] or 0x0, mask )
+
 		-- Write the data
 		if ebitPos <= 32 then
 			self._data[ i_word ] = bor( data, lshift( int, 32 - ebitPos ) )
 			return
 		end
+
 		local overflow = ebitPos - 32 -- [[ 1, 31 ]]
 		self._data[ i_word ] = bor( data, rshift( int, overflow ) )
-		local data = band( rshift( b_mask, overflow ) , self._data[ i_word + 1 ] or 0x0)
-		self._data[ i_word + 1 ] = bor( data, lshift( int, 32 - overflow ))
+
+		data = band( rshift( b_mask, overflow ), self._data[ i_word + 1 ] or 0x0 )
+		self._data[ i_word + 1 ] = bor( data, lshift( int, 32 - overflow ) )
+
 		return self
 	end
 
+	--- @param self BitBuffer
+	--- @param bits number
+	--- @return number # The read data
 	function readraw( self, bits )
-		local rshift, band = rshift, band
```

Comment:
I might put these types of in-function variables back. For some reason Jit will decrease the cycles, if given the functions are within.
I'll do more testing on later.